src/mcp_optimizer/ingestion.py

```
@@ -1746,11 +1773,16 @@ async def ingest_registry(self, toolhive_client: ToolhiveClient) -> None:
         )
 
         # Fetch registry outside transaction (read-only operation)
-        try:
```

same comment as above of maybe it's better to raise a special exception of `ToolHiveNotFound` and handle it in this `try`, instead of deleting the try. However, I think this would work

---

Done